src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs

```diff
@@ -23,21 +23,31 @@ public static class VsHierarchyUtility
         public static string GetProjectPath(IVsHierarchy project)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
-
-            project.GetCanonicalName(VSConstants.VSITEMID_ROOT, out string projectPath);
+            var format = project as IPersistFileFormat;
+            format.GetCurFile(out string projectPath, out uint _);
             return projectPath;
         }
 
-        public static bool IsSupported(IVsHierarchy hierarchy, string projectTypeGuid)
+        public static bool IsSupportedByGuid(IVsHierarchy hierarchy)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
+            ErrorHandler.ThrowOnFailure(hierarchy.GetGuidProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID.VSHPROPID_TypeGuid, out Guid pguid));
+            var guid = pguid.ToString("B");
```

Comment:
Do we need to check if value is `Guid.Empty` just in case?

Reply:
I don't think is likely to fail. 
I think it'd be an overkill to validate this. 

The challenging part is that we're actually calling into native code here, so we can't really get as many validations as we would if everything was strongly typed.